### DISCUSSION TITLE
[login-site] unable to configure it correctly

### DISCUSSION THREAD

**Discussion by Epado** [mia-platform/community#131] [Q&A] [answered]
Hi we are trying to configure the login site and we are encountering some problems, we are following the doc, in particular https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/how_to_enable_auth0 and https://docs.mia-platform.eu/docs/dev_portal/authentication_configuration#endpoints.
We configured the auth0 tenant (a while ago) i'd say successfully cause we are using the same tenant from a different application (leaving in a separated project) and it works fine, so i guess the problem is related to how we configured the endpoints or auth0-client / oauth-login-site within the new project. 

So we added the endpoints /web-login  (/oauth/token and authorize) but the call towards /web-login returns 401, the weird thing, for me at least, is that the authorization service responds with 200 but the api-gateway 401 (i've been using the reqId to filter out the call that reaches the api-gateway). 

I've added some debug information and from the response i get the following info:
- x-debug-proxy-name: wrong-1-GET-/web-login

Do you have any suggestion on what i am doing wrong here ? 
Thanks in advance

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, the `wrong` in the resulted `x-debug-proxy-name` header means that the client-key evaluation by the authorization service is failing; are you providing a client-key that is unknown to the project?

**Reply by davidebianchi** [reply]
The evaluation of the secret is done by the api-gateway, not the authorization-service. This is the reason why the authorization-service respond with 200, and api-gateway with 401.

**Reply by Epado** [reply]
it was declared but for some reason not marked as active. 
Thank you!

**Reply by fredmaggiowski** [reply]
>The evaluation of the secret is done by the api-gateway, not the authorization-service

Yeah this is correct!

**Reply by Epado** [reply]
one further question, just to be sure i've understood what was going on, why was the api-gateway was still checking the secret even if no apiKey was required? because the call had the secret whithin the headers ?

**Reply by fredmaggiowski** [reply]
>because the call had the secret whithin the headers ?

yes, this is the reason. API Key are always taken into account, so if you provide an invalid API Key the request gets rejected even though there was no actual requirement for it.

**Reply by Epado** [reply]
ok thank you again!
